### DISCUSSION TITLE
get_results query doesn't work with diesel schema but works with execute

### DISCUSSION THREAD

**Discussion by Focx-o** [diesel-rs/diesel#3611] [Q&A] [answered]
What does this means?
```
the trait bound `(diesel::sql_types::Nullable<diesel::sql_types::Integer>, diesel::sql_types::Text, diesel::sql_types::Text, diesel::sql_types::Text, diesel::sql_types::Timestamp, diesel::sql_types::Text, diesel::sql_types::Timestamp, diesel::sql_types::Text): load_dsl::private::CompatibleType<models::Employee, Sqlite>` is not satisfied
```
on the basis of this 

```
 let now = Utc::now().to_string();
            let ne = NewEmployee {
                name: &meta["Employee First Name and Last Name"].to_string().replace("\"", ""),
                department: &"",
                software_ids: &"",
                created_on: &now,
                created_by:  & "ab-system",
                updated_on: &now,
                updated_by:  & "ab-system",
            };


         diesel::insert_into(employee::table)
                .values(ne)
                .execute(conn)
                .expect("Error saving new requestor");
        
         // Get the ID of the last inserted row
         let result = employee::dsl::employee.filter(updated_on.eq(now))
         .get_result::<Employee>(conn).unwrap()   <<<==== Here conn showing the above error
     

println!("{:#?}", result);
```

the traits are in order while insert_into.

**Comment by weiznich** [comment]
That's a mismatch between what your query returns and what your `Employee` type expects the query to return. Given the provided information it's hard to tell what's the issue there, but likely that's caused by a mismatch in field order, field types or field count between what your query returns and that struct. Checkout the relevant documentation: https://docs.diesel.rs/2.0.x/diesel/prelude/derive.Queryable.html

**Reply by Focx-o** [reply]
> That's a mismatch between what your query returns and what your `Employee` type expects the query to return. Given the provided information it's hard to tell what's the issue there, but likely that's caused by a mismatch in field order, field types or field count between what your query returns and that struct. Checkout the relevant documentation: https://docs.diesel.rs/2.0.x/diesel/prelude/derive.Queryable.html

If the fields are mismatch it must not have inserted the database. Its Clearly written in the documentation that I can use get_results or get_result once I add the returning clause as a feature.
https://diesel.rs/guides/all-about-inserts.html#the-returning-clause

```
         let result = employee::dsl::employee.filter(updated_on.eq(now))
         .get_result::<Employee>(conn).unwrap()   <<<==== Here conn showing the above error
```

**Reply by weiznich** [reply, marked as answer]
I'm not sure why you feel you must argue about that. At least it's not a constructive way of asking for support. 
The documenation states that you can use `get_results` there, but it doesn't say anything about the requirements on the return types (they are as written above). Additionally your example does not even use that feature but a separate query to load data. 

> If the fields are mismatch it must not have inserted the database. 

You are aware that you would use different structs for inserting and loading in that case? How would you reject the insert part if there is a mismatch in the load part?

Anyway that's my last answer here if you don't provide a reasonable complete example to reproduce your problem and at least try to formulate questions and not requests. That example should at least include the relevant parts of your schema.rs file, definitions of all types that are neither from diesel nor from rust-stdlib and exact version information about which rustc and diesel version you use.

**Reply by Focx-o** [reply]
@weiznich Many apologies. I didn't mean to argue on that. I will be more constructive and reasonable. Thank you for pointing me out. Kind regards.
